tests/test_root_commands.py

```diff
@@ -1,28 +1,30 @@
-
 def test_attr_cmd(hlwm):
     assert hlwm.get_attr('monitors.focus.name') == ''
     hlwm.call('attr')
-    hlwm.callstr('attr tags')
-    hlwm.callstr('attr tags.')
-    hlwm.callstr('attr tags.count')
-    assert hlwm.call('attr', 'tags.count').stdout == hlwm.get_attr('tags.count')
-    hlwm.call_xfail('attr', 'tags.co')
+    hlwm.call('attr tags')
+    hlwm.call('attr tags.')
+    hlwm.call('attr tags.count')
+    assert hlwm.call('attr tags.count').stdout == hlwm.get_attr('tags.count')
+    hlwm.call_xfail('attr tags.co')
+
 
 def test_object_tree(hlwm):
     t1 = hlwm.call('object_tree').stdout.splitlines()
-    t2 = hlwm.call('object_tree', 'theme.').stdout.splitlines()
-    t3 = hlwm.call('object_tree', 'theme.tiling.').stdout.splitlines()
+    t2 = hlwm.call('object_tree theme.').stdout.splitlines()
+    t3 = hlwm.call('object_tree theme.tiling.').stdout.splitlines()
     assert len(t1) > len(t2)
     assert len(t2) > len(t3)
 
+
 def test_sprintf(hlwm):
     cnt = hlwm.get_attr('tags.count')
     wmname = hlwm.get_attr('settings.wmname')
-    printed = hlwm.callstr('sprintf X %s/%s tags.count settings.wmname echo X').stdout
-    hlwm.call_xfail('sprintf X %s/%s tags.count settings.wmname')
-    hlwm.call_xfail('sprintf X %s/%s tags.count')
+    printed = hlwm.call('sprintf X %s/%s tags.count settings.wmname echo X').stdout
+    hlwm.call_xfail('"sprintf X %s/%s tags.count settings.wmname"')
+    hlwm.call_xfail('"sprintf X %s/%s tags.count"')
```

Comment:
These `call_xfail` calls should be without the quotes, because it checks the error handling for lacking arguments or commands

Reply:
Of course, this was wrong before. And I think it supports your point to check also error messages

Reply:
Oops. I didn't look at those tests too closely and just assumed that it is actually supposed to be single string. I'll fix it.